cpu/stm32_common/periph/i2c_1.c

```diff
@@ -0,0 +1,497 @@
+    gpio_init_af(i2c_config[dev].scl_pin, i2c_config[dev].scl_af);
+    gpio_init(i2c_config[dev].sda_pin, GPIO_OD_PU);
+    gpio_init_af(i2c_config[dev].sda_pin, i2c_config[dev].sda_af);
+
+    DEBUG("[i2c] init: configuring device\n");
+    /* set the timing register value from predefined values */
+    i2c_timing_param_t tp = timing_params[i2c_config[dev].speed];
+    uint32_t timing = (( (uint32_t)tp.presc << I2C_TIMINGR_PRESC_Pos) |
+                       ( (uint32_t)tp.scldel << I2C_TIMINGR_SCLDEL_Pos) |
+                       ( (uint32_t)tp.sdadel << I2C_TIMINGR_SDADEL_Pos) |
+                       ( (uint16_t)tp.sclh << I2C_TIMINGR_SCLH_Pos) |
+                       tp.scll);
+    _i2c_init(i2c, timing);
+}
+
+static void _i2c_init(I2C_TypeDef *i2c, uint32_t timing)
+{
+    assert(i2c != NULL);
+
+    /* disable device */
+    i2c->CR1 &= ~(I2C_CR1_PE);
+
+    /* configure analog noise filter */
+    i2c->CR1 |= I2C_CR1_ANFOFF;
+
+    /* configure digital noise filter */
+    i2c->CR1 |= I2C_CR1_DNF;
+
+    /* set timing registers */
+    i2c->TIMINGR = timing;
+
+    /* configure clock stretching */
+    i2c->CR1 &= ~(I2C_CR1_NOSTRETCH);
+
+    /* Clear interrupt */
+    i2c->ICR |= CLEAR_FLAG;
+
+    /* enable device */
+    i2c->CR1 |= I2C_CR1_PE;
+}
+
+int i2c_acquire(i2c_t dev)
+{
+    assert(dev < I2C_NUMOF);
+
+    mutex_lock(&locks[dev]);
+
+    periph_clk_en(i2c_config[dev].bus, i2c_config[dev].rcc_mask);
+
+    return 0;
+}
+
+int i2c_release(i2c_t dev)
+{
+    assert(dev < I2C_NUMOF);
+
+    uint16_t tick = TICK_TIMEOUT;
+
+    while ((i2c_config[dev].dev->ISR & I2C_ISR_BUSY) && tick--) {}
+
+    periph_clk_dis(i2c_config[dev].bus, i2c_config[dev].rcc_mask);
+
+    mutex_unlock(&locks[dev]);
+    return 0;
+}
+
+int i2c_read_bytes(i2c_t dev, uint16_t address, void *data,
+                   size_t length, uint8_t flags)
+{
+    assert(dev < I2C_NUMOF);
+
+    I2C_TypeDef *i2c = i2c_config[dev].dev;
+
+    int ret = 0;
+
+    if (!(flags & I2C_NOSTART)) {
+        DEBUG("[i2c] read_bytes: start transmission\n");
+        /* start reception and send slave address */
+        ret = _start(i2c, address, length, I2C_FLAG_READ, flags);
+        if (ret < 0) {
+            return ret;
+        }
+    }
+
+    DEBUG("[i2c] read_bytes: read the data\n");
+    /* read the data bytes */
+    ret = _read(i2c, data, length);
+    if (ret < 0) {
+        DEBUG("[i2c] read_bytes: error while reading\n");
+        return ret;
+    }
+
+    if (!(flags & I2C_NOSTOP)) {
+        DEBUG("[i2c] read_bytes: end transmission\n");
+        ret = _stop(i2c);
+        if (ret < 0) {
+            return ret;
+        }
+    }
+
+    return ret;
+}
+
+int i2c_read_regs(i2c_t dev, uint16_t address, uint16_t reg, void *data,
+                  size_t length, uint8_t flags)
+{
+    assert(dev < I2C_NUMOF);
+
+    DEBUG("[i2c] read_regs: addr: %04X, reg: %04X\n", address, reg);
+
+    if (flags & I2C_REG16) {
```

Comment:
add I2C_ADDR10 as an unsupported feature please. We still cannot test this feature without a specific sensor to test it.

Reply:
Agreed, fixed.